google/cloud/spanner/client.h

```diff
@@ -267,10 +267,51 @@ class Client {
   QueryResult ExecuteQuery(QueryPartition const& partition);
   //@}
 
+  //@{
+  /**
+   * Profiles a SQL query.
+   *
+   * Profiling executes the query, provides the resulting rows, `ExecutionPlan`,
+   * and execution statistics.
```

Comment:
Should we say something about the caller is responsible to consume all the returned rows form the query, and the profile information is not available until all the rows have been consumed?

Reply:
Done.